addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -556,24 +560,66 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
             }
         }
 
+        /**
+         * Returns a hexadecimal encoded MD5 hash for the input String.
+         *
+         * @param data
+         * @return
+         */
+        private String getMD5Hash(String data) {
+            String result = null;
+            try {
+                MessageDigest digest = MessageDigest.getInstance("MD5");
+                byte[] hash = digest.digest(data.getBytes("UTF-8"));
+                return bytesToHex(hash); // make it printable
+            } catch (Exception ex) {
+                ex.printStackTrace();
+            }
+            return result;
+        }
+
+        /**
+         * Use javax.xml.bind.DatatypeConverter class in JDK to convert byte array
+         * to a hexadecimal string. Note that this generates hexadecimal in upper case.
+         *
+         * @param hash
+         * @return
+         */
+        private String bytesToHex(byte[] hash) {
+            return DatatypeConverter.printHexBinary(hash);
+        }
+
         private String fetchUrl(String messagePart, final String mac) {
             // default url for cover art. Works all the time except for some particular case
             String url = "http://" + host + ":" + webport + "/music/current/cover.jpg?player=" + encode(mac);
+            // return default url when no artwork_url and no coderid
+            if (messagePart == null) {
+                return url;
+            }
             // if messagePart start with "artwork_url:http://", default url works
-            if (messagePart != null && messagePart.startsWith("artwork_url%3A")
-                    && !messagePart.startsWith("artwork_url%3Ahttp%3A%2F%2F")) {
-                url = messagePart.substring("artwork_url%3A".length());
-                // example of particular case.
-                // this web radio : http://broadcast.infomaniak.net/tsfjazz-high.mp3
-                // default url return error 404
-                // messagePart = artwork_url%3Ahtml%2Fimages%2Fradio.png (artwork_url:html/images/radio.png)
-                // working url : "http://" + host + ":" + webport + "/" + "html/images/radio.png";
-                if (url.startsWith("%2F")) {
-                    url = "http://" + host + ":" + webport + decode(url);
+            if (messagePart.startsWith("artwork_url%3A")) {
+                if (messagePart.startsWith("artwork_url%3Ahttp%3A%2F%2F")
+                        || messagePart.startsWith("artwork_url%3Ahttps%3A%2F%2F")) {
+                    // add hash parameter to prevent problem with image caching
```

Comment:
Maybe elaborate a little more about the caching problems, so explain that the `messagePart` is stable as long as the same song is playing so that it makes sure that the cache is used as long as we have the same `messagePart`.

At least that is what I understood  but if that is not the case, then I see an extra reason to document.